src/server/routes/controllers/email.js

```diff
@@ -0,0 +1,119 @@
+const responseManager = require('./utilities/responseManager')
+const config = require('../../config.json')
+const nodemailer = require('nodemailer')
+const logger = require('../../logger')
+const crypto = require('crypto')
+const _ = require('lodash')
+
+const testAccount = {
+    email: process.env.EMAIL || config.email.email,
+    password: process.env.EMAIL_PASSWORD || config.email.password,
+    host: process.env.EMAIL_HOST || config.email.host
+}
+
+const SMTP_OPTIONS = { //TODO: Add SMTPS
+    host: `smtp.${testAccount.host}`,
+    port: 587,
+    secure: false,
+    auth: {
+        user: testAccount.email,
+        pass: testAccount.password
+    }
+}
+
+async function prepareTransporter(){
+    let transporter = nodemailer.createTransport(SMTP_OPTIONS)
+
+    await transporter.verify((err) => {
+        if(err) return console.log(err)
+        logger.info('Email server is ready to take our messages')
+    })
+
+    return transporter
+}
+
+function createToken(){
+    const token = crypto.randomBytes(20).toString('hex')
+    return token
+}
+
+function setMailOptions(recvAddress, subject, body){
+    return {
+        from: '"Lokim Messenger Services" <lokim.messenger@mail.com>',
+        to: recvAddress,
+        subject,
+        text: body
+    }
+}
+
+function sendMail(transporter = prepareTransporter()){
+    return (mailOptions) => {
+        for(let key in mailOptions){
```

Comment:
What do you think about
```
_.forEach(mailOptions, optionValue => {
    if (_.isEmpty(optionValue)) throw new Error('Mail options cannot be unset.')
})
```

Reply:
Alternative solution
```
const unsetValuesList = _.filter(mailOptions, _.isEmpty)
if (!_.isEmpty(unsetValuesList)) throw new Error('Mail options cannot be unset.')
```

Reply:
I'll try out both of these and implement them if they work